### ISSUE TITLE
 BlobCache.LocalMachine.GetAllKeys() very slow with 400 3MB byte[] objects..

### ISSUE THREAD

**Issue by mms-** [reactiveui/Akavache#284]
Slow is about  30sec on every sequential call. It seems like it's loading all the object data vs just getting the keys? 

This is on a Lumia 830, Win 10. It's only 1.2GB of data for all the objects, but perhaps Akavache is meant for simple string storage only? In that case why call it BlobCache if an image sized blob can't be handled in any significant number? I don't think sqlite is the cause?

**Comment by flagbug** [comment]
I've just looked through the `GetAllKeys` path, and it should correctly select only the keys in the SQLite table: https://github.com/akavache/Akavache/blob/5e94aed82c6caa9e006a0232ded74d3d00bfe9f2/Akavache.Sqlite3/Operations.cs#L454, so it seems like something else is going on here

**Comment by mms-** [comment]
Yes very strange. It's not exactly a slow phone. I will do more tests on other devices.

**Comment by flagbug** [comment]
This may be because there is no index over the Expiration database field, see #285

**Comment by glennawatson** [comment]
Thank you for reporting this. Closing as this has been optimized in Akavache V11.

In V11, `GetAllKeys()` only selects the `Id` column from the SQLite database — it never loads the `Value` (blob data) column. The query is equivalent to `SELECT Id FROM CacheEntry WHERE ...`, so performance is independent of blob sizes.

The old V10 implementation may have been loading full rows including blob data, which would explain the 30-second delays with large objects. This is no longer the case.

If you experience performance issues with V11, please open a new issue.
